tools/upload_to_anaconda_staging.sh

```diff
@@ -30,7 +28,7 @@ upload_wheels() {
 
         anaconda -t $ANACONDA_SCIENTIFIC_PYTHON_UPLOAD upload \
                 --no-progress --force -u scientific-python-nightly-wheels \
-                dist/scipy-openblas*.whl
+                dist/scipy_openblas*.whl
```

Comment:
This won't work. The upload will fail trying to upload to a project named `https://anaconda.org/scientific-python-nightly-wheels/scipy_openblas32/` (with a `_`) where the project is named `https://anaconda.org/scientific-python-nightly-wheels/scipy-openblas32/` (with a '-'). Look at the uploads to `https://anaconda.org/multibuild-wheels-staging/`, where I created both options. Only after I did the wheel renaming would the upload got correctly to the `-` project.

Reply:
@mattip No. You are confusing project name with wheel filename. These are not the same.

Please reread the body of this PR that I opened where I clearly show this.

Reply:
Sorry, typing quickly as going offline, but here's maybe a more helpful example of what I mean form the PR body in the logs:

```
+ anaconda --token *** upload --force --user scientific-python-nightly-wheels --label main dist/test_package-0.0.1-py3-none-any.whl
Using Anaconda API: https://api.anaconda.org/
Using "scientific-python-nightly-wheels" as upload username
Processing "dist/test_package-0.0.1-py3-none-any.whl"
Detecting file type...
File type is "Standard Python"
Extracting standard python attributes for upload
Creating package "test-package"
Creating release "0.0.1"
Uploading file "scientific-python-nightly-wheels/test-package/0.0.1/test_package-0.0.1-py3-none-any.whl"

  0%|          | 0.00/1.27k [00:00<?, ?B/s]
6.78kB [00:00, 28.6kB/s]                   
6.78kB [00:00, 17.7kB/s]
Upload complete

standard python located at:
  https://anaconda.org/scientific-python-nightly-wheels/test-package
```

(this project gets deleted at the end of the CI run, so it isn't there now)

* project-name in `pyproject.toml`: `test-package`
* wheel filename`: `test_package-0.0.1-py3-none-any.whl`
* Project name on Anaconda Cloud: `test-package`

Reply:
c.f. https://github.com/MacPython/openblas-libs/pull/140#issuecomment-1952790915